### PR TITLE
feat(auth): add DID VC datatypes and sample testing

### PR DESCRIPTION
**Pull request by saitofun**

### CODE REVIEW

File: auth/didvc/vc_test.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package didvc_test
```
Comment (CoderZhi) [inline auth/didvc/vc_test.go:1]: two different package names in same directory?

File: auth/didvc/vc_test.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/machinefi/sprout/auth/didvc"
+	"github.com/stretchr/testify/require"
+	"io"
+	"net/http"
+	"testing"
+	"time"
```
Comment (CoderZhi) [inline auth/didvc/vc_test.go:11]: group

File: auth/didvc/types.go [hunk 3]
```diff
@@ -0,0 +1,87 @@
+// this file defines Verifiable Credential datatype.
+// ref: https://w3c-ccg.github.io/vc-api/
+package didvc
```
Comment (CoderZhi) [inline auth/didvc/types.go:3]: did

**Review by CoderZhi — COMMENTED** [review]
fix unit test

**Review by huangzhiran — APPROVED** [review]

**Pull request merged**